### ISSUE TITLE
Ripple of navigation button gets interrupted when loading a new view

### ISSUE THREAD

**Issue by lbjones** [material-components/material-components-web#1381]
For the life of me I can't figure out why the ripple doesn't work for me any more. 

I just upgraded from 0.16 to 0.21.1 and using webpack to compile ES2015 with vuejs framework.

I tried in vain to create a jsbin, but I'm not sure if it can be done to import the package.  So here is the basics of my code:

SCSS:
```
@import "@material/button/mdc-button";
```

HTML:
```
<button id="mybutton" class="mdc-button">
    Button
</button>
```
JS
```
import {MDCRipple} from '@material/ripple';
const surface = document.querySelector('#mybutton');
const ripple = new MDCRipple(surface);
console.log(ripple);
```

In the console, the ripple seems to have been instantiated correctly.  But when I click it there is no ripple!  Only the css flash is occurring.

**Comment by acdvorak** [comment]
Thanks for filing this issue!

Could you possibly create a small GitHub repo or Gist with the minimum amount of code needed to expose the bug?

Does the issue still occur if you don't use Vue?

**Comment by lbjones** [comment]
Ok I created the repo here:  https://github.com/lbjones/vue-ripple-test

I found that the issue is slightly different than initially thought.  The ripple works just fine.  But the problem is that animation is disrupted by the router event when I navigate to a new view within my single page application.  Also the ripple no longer works at all in the router buttons after the first navigation. 

I use the buttons in the header and would like the animation to not be disrupted.  You'll also see that the destroy method is not being called as expected, so I'm not sure what is causing the problem.  

You'll also see in this sample repo that another button from a project called Keen-Ui is provided as a side by side comparison, and it works flawlessly.

Any insight would be much appreciated!

**Comment by acdvorak** [comment]
This sounds like a framework issue, but I'll take a look at your repo first to confirm (thanks for creating that!).

**Comment by acdvorak** [comment]
TL;DR: You need to re-instantiate `MDCRipple` objects after client-side navigation occurs. You're already doing this for the button in the content area (`#b4`), but it looks like `App.vue`'s `mounted()` method isn't getting called after navigating.

Running the example repo you provided (thanks!) confirmed my suspicion: The `mdc-ripple-upgraded` class is getting removed from the anchor element whenever there's a client-side navigation, and it's not getting re-added after navigation. If you manually add the class back, the ripple suddenly works again.

I don't know anything about Vue, so I can't say for sure why that class is getting removed, but I'm guessing it's because Vue is re-rendering the DOM from a template, and that template _doesn't_ have the `mdc-ripple-upgraded` class (which is correct). The class is added dynamically at runtime by the `MDCRipple` JS, so I don't recommend hard-coding it in your Vue template, but the `MDCRipple` constructor _should_ be adding the class when it calls the foundation's `init()` method.

It looks like the `MDCRipple` constructor isn't getting called for the nav buttons at the top of the page, but it _is_ getting called for the "(MDC) STATIC HOME BUTTON" in the content area.

Closing this issue as it doesn't appear to be an MDC-Web bug, but feel free to reopen if you disagree 🙂

**Comment by lbjones** [comment]
Hey thanks for looking into it!  I thought maybe the destroy method was being called, but that is not the case.  It's kind of a mystery then what is causing the `mdc-ripple-upgraded` class to be removed.  If I figure it out I'll post a solution here.

**Comment by jjstanton** [comment]
Hi @lbjones. I just came across this issue today using Vue 3 and MDC. 
Did you ever manage to find a solution for this?

**Comment by lynden-jones** [comment]
I believe I figured out a workaround.  I was using tailwindcss and the preflight was messing it up.  I overrode the preflight css with something like this:

```
/**
 * Allow adding a border to an element by just adding a border-width.
 *
 * By default, the way the browser specifies that an element should have no
 * border is by setting it's border-style to `none` in the user-agent
 * stylesheet.
 *
 * In order to easily add borders to elements by just setting the `border-width`
 * property, we change the default border-style for all elements to `solid`, and
 * use border-width to hide them instead. This way our `border` utilities only
 * need to set the `border-width` property instead of the entire `border`
 * shorthand, making our border utilities much more straightforward to compose.
 *
 * https://github.com/tailwindcss/tailwindcss/pull/116
 */
* {
  border-width: 0;
  // border-style: solid;  // this line messes with mdc-button ripple
  border-color: config('borderColors.default', currentColor);
}
*::before, *::after {
  border-width: 0;
  border-style: solid;
  border-color: config('borderColors.default', currentColor);
}
```

**Comment by jjstanton** [comment]
After a lot of trial and error I believe I have also found a solution for this issue. 

From what I understand, Vue Routers `router-link` will automatically add the [active classes](https://router.vuejs.org/api/#active-class) `router-link-active` & `router-link-exact-active` depending on the active route. When this happens, I believe Vue Router stores the elements initial state before adding the active classes and returns that initial state after the route and the route-link no longer match.

Since Vue Router is instantiated before any MDC ripple elements, the `mdc-ripple-upgraded` class will not be in the elements initial state and will not be re-applied, giving the appearance of the `mdc-ripple-upgraded` class being removed after navigating away from its route. 

The solution that works for me is by using router-links v-slot API to set the active classes on a parenting span element while still allowing the click event to be attached to the element I want the ripple affect to instantiated to, in my case the `mdc-list-item` element. 

HTML:
```
    <router-link v-if="!menu.subItems" :to="menu.route" custom v-slot="{ navigate, href, isActive, isExactActive }">
        <span :class="[isActive && 'router-link-active', isExactActive && 'mdc-list-item--selected']">
            <a class="mdc-list-item" :href="href" @click="navigate">
                <span class="mdc-list-item__ripple"></span>
                <i v-if="!menu.fontAwesome" class="material-icons mdc-list-item__graphic" aria-hidden="true">{{ menu.icon }}</i>
                <i v-else class="mdc-list-item__graphic" :class="menu.icon" aria-hidden="true"></i>
                <span class="mdc-list-item__text">{{ menu.name }}</span>
            </a>
        </span>
    </router-link>
```

JS:
```
import { ripple } from "material-components-web";
export default {
        mounted() {
            let menuItem = document.querySelector(".mdc-list-item");
            new ripple.MDCRipple(menuItem);
        }
};
```

SCSS:
```
    span.mdc-list-item--selected :not(.mdc-list-item--disabled):not(.mdc-list-group-header).mdc-list-item.mdc-ripple-upgraded,
    .mdc-list-group--items span.mdc-list-item--selected .mdc-list-item.mdc-ripple-upgraded {
        --mdc-ripple-fg-opacity: 0.3;
        background-color: #325887;
        transition: background-color 225ms linear;
    }
```
